src/devices/Ft4222/Ft4222I2c.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Device.Ft4222;
+using System.IO;
+
+namespace System.Device.I2c
+{
+    /// <summary>
+    /// FT4222 I2C Device
+    /// </summary>
+    public class Ft4222I2c : I2cDevice
+    {
+        private I2cConnectionSettings _settings;
+        private IntPtr _ftHandle = new IntPtr();
+
+        /// <summary>
+        /// Store the FTDI Device Information
+        /// </summary>
+        public DeviceInformation DeviceInformation { get; internal set; }
+
+        /// <summary>
+        /// Create a FT4222 I2C Device
+        /// </summary>
+        /// <param name="settings">I2C Connection Settings</param>
+        public Ft4222I2c(I2cConnectionSettings settings)
+        {
+            _settings = settings;
+            // Check device
+            var devInfos = FtCommon.GetDevices();
+            // Select the one from bus Id
+            DeviceInformation = devInfos[_settings.BusId];            
+            
+            // Open device
+            var ftStatus = FtFunction.FT_OpenEx(DeviceInformation.LocId, FtOpenType.OpenByLocation, ref _ftHandle);
+
+            if (ftStatus != FtStatus.Ok)
+                throw new IOException($"Failed to open device {DeviceInformation.Description}, status: {ftStatus}");
+
+            // Set the clock
+            FtClockRate ft4222Clock = FtClockRate.Clock24MHz;
+
+            ftStatus = FtFunction.FT4222_SetClock(_ftHandle, ft4222Clock);
+            if (ftStatus != FtStatus.Ok)
+                throw new IOException($"Failed set clock rate {ft4222Clock} on device: {DeviceInformation.Description}, status: {ftStatus}");
+
+            // Set the device as I2C Master
+            ftStatus = FtFunction.FT4222_I2CMaster_Init(_ftHandle, 400);
```

Comment:
what's 400?

Reply:
Good catch, the I2C Frequency in kbps. Added as a constant. Btw, I didn't find a way to pass this in the I2cConnectionSettings, is there somewhere a way to change it while constructing the I2C class?